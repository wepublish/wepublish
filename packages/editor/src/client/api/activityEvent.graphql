fragment FullActivityEvent on ActivityEvent {
  id
  date
  eventType
  summary
  creator
}

query RecentActivity {
    activityEvents {
        ...FullActivityEvent
    }
}