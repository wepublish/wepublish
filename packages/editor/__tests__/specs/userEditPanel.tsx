import React from 'react'
import {UserEditPanel} from '../../src/client/panel/userEditPanel'
import {UserDocument, CreateUserDocument, UserRoleListDocument} from '../../src/client/api'
import {mount} from 'enzyme'

import {updateWrapper} from '../utils'
import {act} from 'react-dom/test-utils'
import {MockedProvider as MockedProviderBase} from '@apollo/client/testing'

const MockedProvider = MockedProviderBase as any

const userRoleListDocumentQuery = {
  request: {
    query: UserRoleListDocument,
    variables: {
      first: 200
    }
  },
  result: () => {
    return {
      data: {
        userRoles: {
          nodes: [
            {
              __typename: 'UserRole',
              id: 'roleId1',
              name: 'Role 1',
              description: 'Description for role 1',
              systemRole: false,
              permissions: []
            },
            {
              __typename: 'UserRole',
              id: 'roleId2',
              name: 'Role 2',
              description: 'Description for role 2',
              systemRole: false,
              permissions: []
            }
          ],
          pageInfo: {
            hasNextPage: false,
            hasPreviousPage: false
          },
          totalCount: 2
        }
      }
    }
  }
}

const userDocumentQuery = {
  request: {
    query: UserDocument,
    variables: {
      id: 'fakeId3'
    }
  },
  result: () => ({
    data: {
      user: {
        __typename: 'User',
        id: 'fakeId3',
        name: 'Peter Parker',
        email: 'peter@parker.com',
        roles: [
          {
            __typename: 'UserRole',
            id: 'roleId1',
            name: 'Role 1',
            description: 'Description for role 1',
            systemRole: false,
            permissions: []
          }
        ]
      }
    }
  })
}

describe('User Edit Panel', () => {
  test('should render', async () => {
    const mocks = [userRoleListDocumentQuery]
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <UserEditPanel />
      </MockedProvider>
    )
    await updateWrapper(wrapper, 100)

    const panel = wrapper.find('UserEditPanel')
    expect(panel).toMatchSnapshot()
  })

  test('should render with ID', async () => {
    const mocks = [userDocumentQuery, userRoleListDocumentQuery]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={true}>
        <UserEditPanel id={'fakeId3'} />
      </MockedProvider>
    )
    await updateWrapper(wrapper, 100)

    const panel = wrapper.find('UserEditPanel')
    expect(panel).toMatchSnapshot()
  })

  test('should allow user role to be added', async () => {
    const mocks = [userRoleListDocumentQuery]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <UserEditPanel />
      </MockedProvider>
    )
    await updateWrapper(wrapper, 100)

    wrapper.find('a[name="userList.panels.userRoles"]').simulate('click')
    wrapper.find('div[role="menuitem"]').last().simulate('click')

    const panel = wrapper.find('UserEditPanel')
    expect(panel).toMatchSnapshot()
  })

  test('should allow user role to be removed', async () => {
    const mocks = [userDocumentQuery, userRoleListDocumentQuery]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={true}>
        <UserEditPanel id={'fakeId3'} />
      </MockedProvider>
    )
    await updateWrapper(wrapper, 100)

    wrapper.find('a[name="userList.panels.userRoles"]').simulate('click')
    wrapper.find('div.rs-checkbox-checked').simulate('click')

    const panel = wrapper.find('UserEditPanel')
    expect(panel).toMatchSnapshot()
  })

  test('should allow a new user to be created', async () => {
    const user = {
      name: 'Testing MÃ¼ller',
      email: 'testing@mueller.com',
      active: true,
      properties: [],
      roleIDs: [],
      password: 'superSecret'
    }

    const mocks = [
      {
        request: {
          query: CreateUserDocument,
          variables: {
            input: {
              name: user.name,
              preferredName: undefined,
              email: user.email,
              active: user.active,
              properties: user.properties,
              roleIDs: user.roleIDs
            },
            password: user.password
          }
        },
        result: () => ({
          data: {
            user: {
              __typename: 'User',
              id: 'fakeId4',
              name: user.name,
              email: user.email,
              roles: user.roleIDs,
              active: user.active,
              properties: user.properties
            }
          }
        })
      },
      userRoleListDocumentQuery
    ]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <UserEditPanel />
      </MockedProvider>
    )
    await updateWrapper(wrapper, 100)

    act(() => {
      wrapper
        .find('input[name="userList.panels.name"]')
        .simulate('change', {target: {value: user.name}})
      wrapper
        .find('input[name="userList.panels.email"]')
        .simulate('change', {target: {value: user.email}})
      wrapper
        .find('input[name="userList.panels.password"]')
        .simulate('change', {target: {value: user.password}})
    })

    wrapper.find('button[className="rs-btn rs-btn-primary"]').simulate('click')

    const panel = wrapper.find('UserEditPanel')
    expect(panel).toMatchSnapshot()
  })
})
