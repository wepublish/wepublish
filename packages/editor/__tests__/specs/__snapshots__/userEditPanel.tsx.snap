// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Edit Panel should allow a new user to be created 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -35,13 +35,14 @@
        </h4>
        <div
          class=\\"rs-drawer-actions\\"
        >
          <button
-           aria-disabled=\\"false\\"
-           class=\\"rs-btn rs-btn-primary\\"
+           aria-disabled=\\"true\\"
+           class=\\"rs-btn rs-btn-primary rs-btn-disabled\\"
            data-testid=\\"saveButton\\"
+           disabled=\\"\\"
            type=\\"submit\\"
          >
            userList.panels.create
            <span
              class=\\"rs-ripple-pond\\"
@@ -92,10 +93,11 @@
              >
                <input
                  aria-describedby=\\"firstName-help-text\\"
                  aria-labelledby=\\"firstName-control-label\\"
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  id=\\"firstName\\"
                  name=\\"firstName\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
@@ -117,14 +119,15 @@
              >
                <input
                  aria-describedby=\\"name-help-text\\"
                  aria-labelledby=\\"name-control-label\\"
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  id=\\"name\\"
                  name=\\"name\\"
                  type=\\"text\\"
-                 value=\\"\\"
+                 value=\\"Testing MÃ¼ller\\"
                />
              </div>
            </div>
            <div
              class=\\"rs-form-group\\"
@@ -138,10 +141,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"preferredName\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -162,14 +166,15 @@
              >
                <input
                  aria-describedby=\\"email-help-text\\"
                  aria-labelledby=\\"email-control-label\\"
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  id=\\"email\\"
                  name=\\"email\\"
                  type=\\"text\\"
-                 value=\\"\\"
+                 value=\\"testing@mueller.com\\"
                />
              </div>
            </div>
            <div
              class=\\"rs-form-group\\"
@@ -183,10 +188,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"company\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -203,10 +209,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"streetAddress\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -223,10 +230,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"streetAddress2\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -243,10 +251,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"zipCode\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -263,10 +272,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"city\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -283,10 +293,11 @@
              <div
                class=\\"rs-form-control rs-form-control-wrapper\\"
              >
                <input
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  name=\\"country\\"
                  type=\\"text\\"
                  value=\\"\\"
                />
              </div>
@@ -299,18 +310,19 @@
                class=\\"rs-form-control-label\\"
              >
                userList.panels.active
              </label>
              <label
-               class=\\"rs-toggle rs-toggle-checked\\"
+               class=\\"rs-toggle rs-toggle-checked rs-toggle-disabled\\"
              >
                <input
                  aria-checked=\\"true\\"
-                 aria-disabled=\\"false\\"
+                 aria-disabled=\\"true\\"
                  aria-label=\\"Open\\"
                  checked=\\"\\"
                  class=\\"rs-toggle-input\\"
+                 disabled=\\"\\"
                  role=\\"switch\\"
                  type=\\"checkbox\\"
                />
                <span
                  class=\\"rs-toggle-presentation\\"
@@ -337,14 +349,15 @@
              >
                <input
                  aria-describedby=\\"password-help-text\\"
                  aria-labelledby=\\"password-control-label\\"
                  class=\\"rs-input\\"
+                 disabled=\\"\\"
                  id=\\"password\\"
                  name=\\"password\\"
                  type=\\"password\\"
-                 value=\\"\\"
+                 value=\\"superSecret\\"
                />
              </div>
            </div>
            <div
              class=\\"rs-form-group\\""
`;

exports[`User Edit Panel should allow user role to be added 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -354,11 +354,11 @@
                class=\\"rs-form-control-label\\"
              >
                userList.panels.userRoles
              </label>
              <div
-               class=\\"rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-cleanable rs-picker-block rs-picker-countable\\"
+               class=\\"rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-has-value rs-picker-cleanable rs-picker-block rs-picker-countable\\"
              >
                <div
                  aria-expanded=\\"false\\"
                  aria-haspopup=\\"listbox\\"
                  class=\\"rs-picker-toggle rs-btn rs-btn-default\\"
@@ -368,17 +368,54 @@
                  <input
                    aria-hidden=\\"true\\"
                    class=\\"rs-picker-toggle-textbox rs-picker-toggle-read-only\\"
                    readonly=\\"\\"
                    tabindex=\\"-1\\"
-                   value=\\"\\"
+                   value=\\"roleId2\\"
                  />
+                 <span
+                   class=\\"rs-picker-toggle-value\\"
+                 >
+                   <span
+                     class=\\"rs-picker-value-list\\"
+                     title=\\"Role 2\\"
+                   >
                      <span
-                   aria-placeholder=\\"Select\\"
-                   class=\\"rs-picker-toggle-placeholder\\"
+                       class=\\"rs-picker-value-item\\"
+                     >
+                       Role 2
+                     </span>
+                   </span>
+                   <span
+                     class=\\"rs-picker-value-count\\"
+                     title=\\"1\\"
+                   >
+                     1
+                   </span>
+                 </span>
+                 <span
+                   aria-label=\\"Clear\\"
+                   class=\\"rs-picker-toggle-clean rs-btn-close\\"
+                   role=\\"button\\"
+                   tabindex=\\"-1\\"
+                   title=\\"Clear\\"
+                 >
+                   <svg
+                     aria-hidden=\\"true\\"
+                     aria-label=\\"close\\"
+                     class=\\"rs-icon\\"
+                     data-category=\\"application\\"
+                     fill=\\"currentColor\\"
+                     focusable=\\"false\\"
+                     height=\\"1em\\"
+                     viewBox=\\"0 0 16 16\\"
+                     width=\\"1em\\"
                    >
-                   Select
+                     <path
+                       d=\\"M2.784 2.089l.069.058 5.146 5.147 5.146-5.147a.5.5 0 01.765.638l-.058.069L8.705 8l5.147 5.146a.5.5 0 01-.638.765l-.069-.058-5.146-5.147-5.146 5.147a.5.5 0 01-.765-.638l.058-.069L7.293 8 2.146 2.854a.5.5 0 01.638-.765z\\"
+                     />
+                   </svg>
                  </span>
                  <svg
                    aria-hidden=\\"true\\"
                    aria-label=\\"angle down\\"
                    class=\\"rs-picker-toggle-caret rs-icon\\""
`;

exports[`User Edit Panel should allow user role to be removed 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -363,11 +363,11 @@
                class=\\"rs-form-control-label\\"
              >
                userList.panels.userRoles
              </label>
              <div
-               class=\\"rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-has-value rs-picker-cleanable rs-picker-block rs-picker-countable\\"
+               class=\\"rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-cleanable rs-picker-block rs-picker-countable\\"
              >
                <div
                  aria-expanded=\\"false\\"
                  aria-haspopup=\\"listbox\\"
                  class=\\"rs-picker-toggle rs-btn rs-btn-default\\"
@@ -377,54 +377,17 @@
                  <input
                    aria-hidden=\\"true\\"
                    class=\\"rs-picker-toggle-textbox rs-picker-toggle-read-only\\"
                    readonly=\\"\\"
                    tabindex=\\"-1\\"
-                   value=\\"roleId1\\"
+                   value=\\"\\"
                  />
-                 <span
-                   class=\\"rs-picker-toggle-value\\"
-                 >
-                   <span
-                     class=\\"rs-picker-value-list\\"
-                     title=\\"Role 1\\"
-                   >
                  <span
-                       class=\\"rs-picker-value-item\\"
-                     >
-                       Role 1
-                     </span>
-                   </span>
-                   <span
-                     class=\\"rs-picker-value-count\\"
-                     title=\\"1\\"
-                   >
-                     1
-                   </span>
-                 </span>
-                 <span
-                   aria-label=\\"Clear\\"
-                   class=\\"rs-picker-toggle-clean rs-btn-close\\"
-                   role=\\"button\\"
-                   tabindex=\\"-1\\"
-                   title=\\"Clear\\"
-                 >
-                   <svg
-                     aria-hidden=\\"true\\"
-                     aria-label=\\"close\\"
-                     class=\\"rs-icon\\"
-                     data-category=\\"application\\"
-                     fill=\\"currentColor\\"
-                     focusable=\\"false\\"
-                     height=\\"1em\\"
-                     viewBox=\\"0 0 16 16\\"
-                     width=\\"1em\\"
+                   aria-placeholder=\\"Select\\"
+                   class=\\"rs-picker-toggle-placeholder\\"
                  >
-                     <path
-                       d=\\"M2.784 2.089l.069.058 5.146 5.147 5.146-5.147a.5.5 0 01.765.638l-.058.069L8.705 8l5.147 5.146a.5.5 0 01-.638.765l-.069-.058-5.146-5.147-5.146 5.147a.5.5 0 01-.765-.638l.058-.069L7.293 8 2.146 2.854a.5.5 0 01.638-.765z\\"
-                     />
-                   </svg>
+                   Select
                  </span>
                  <svg
                    aria-hidden=\\"true\\"
                    aria-label=\\"angle down\\"
                    class=\\"rs-picker-toggle-caret rs-icon\\""
`;

exports[`User Edit Panel should render 1`] = `
<DocumentFragment>
  <form
    class="rs-form rs-form-vertical rs-form-fluid"
    style="height: 100%;"
  >
    <div
      class="rs-drawer-header"
    >
      <span
        aria-label="Close"
        class="rs-drawer-header-close rs-btn-close"
        role="button"
        title="Close"
      >
        <svg
          aria-hidden="true"
          aria-label="close"
          class="rs-icon"
          data-category="application"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M2.784 2.089l.069.058 5.146 5.147 5.146-5.147a.5.5 0 01.765.638l-.058.069L8.705 8l5.147 5.146a.5.5 0 01-.638.765l-.069-.058-5.146-5.147-5.146 5.147a.5.5 0 01-.765-.638l.058-.069L7.293 8 2.146 2.854a.5.5 0 01.638-.765z"
          />
        </svg>
      </span>
      <h4
        class="rs-drawer-title"
      >
        userList.panels.createUser
      </h4>
      <div
        class="rs-drawer-actions"
      >
        <button
          aria-disabled="false"
          class="rs-btn rs-btn-primary"
          data-testid="saveButton"
          type="submit"
        >
          userList.panels.create
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </button>
        <button
          class="rs-btn rs-btn-subtle"
          type="button"
        >
          userList.panels.close
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="rs-drawer-body"
    >
      <div
        class="rs-panel"
      >
        <div
          class="rs-panel-body"
          role="region"
        >
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="firstName"
              id="firstName-control-label"
            >
              userList.panels.firstName
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="firstName-help-text"
                aria-labelledby="firstName-control-label"
                class="rs-input"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="name"
              id="name-control-label"
            >
              userList.panels.name*
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="name-help-text"
                aria-labelledby="name-control-label"
                class="rs-input"
                id="name"
                name="name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.preferredName
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="preferredName"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="email"
              id="email-control-label"
            >
              userList.panels.email*
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="email-help-text"
                aria-labelledby="email-control-label"
                class="rs-input"
                id="email"
                name="email"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.company
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="company"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.streetAddress
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="streetAddress"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.streetAddress2
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="streetAddress2"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.zipCode
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="zipCode"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.city
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="city"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.country
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="country"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.active
            </label>
            <label
              class="rs-toggle rs-toggle-checked"
            >
              <input
                aria-checked="true"
                aria-disabled="false"
                aria-label="Open"
                checked=""
                class="rs-toggle-input"
                role="switch"
                type="checkbox"
              />
              <span
                class="rs-toggle-presentation"
              >
                <span
                  class="rs-toggle-inner"
                />
              </span>
            </label>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="password"
              id="password-control-label"
            >
              userList.panels.password*
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="password-help-text"
                aria-labelledby="password-control-label"
                class="rs-input"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.userRoles
            </label>
            <div
              class="rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-cleanable rs-picker-block rs-picker-countable"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="rs-picker-toggle rs-btn rs-btn-default"
                role="combobox"
                tabindex="0"
              >
                <input
                  aria-hidden="true"
                  class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                  readonly=""
                  tabindex="-1"
                  value=""
                />
                <span
                  aria-placeholder="Select"
                  class="rs-picker-toggle-placeholder"
                >
                  Select
                </span>
                <svg
                  aria-hidden="true"
                  aria-label="angle down"
                  class="rs-picker-toggle-caret rs-icon"
                  data-category="legacy"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 21 32"
                  width="1em"
                >
                  <path
                    d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`User Edit Panel should render with ID 1`] = `
<DocumentFragment>
  <form
    class="rs-form rs-form-vertical rs-form-fluid"
    style="height: 100%;"
  >
    <div
      class="rs-drawer-header"
    >
      <span
        aria-label="Close"
        class="rs-drawer-header-close rs-btn-close"
        role="button"
        title="Close"
      >
        <svg
          aria-hidden="true"
          aria-label="close"
          class="rs-icon"
          data-category="application"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M2.784 2.089l.069.058 5.146 5.147 5.146-5.147a.5.5 0 01.765.638l-.058.069L8.705 8l5.147 5.146a.5.5 0 01-.638.765l-.069-.058-5.146-5.147-5.146 5.147a.5.5 0 01-.765-.638l.058-.069L7.293 8 2.146 2.854a.5.5 0 01.638-.765z"
          />
        </svg>
      </span>
      <h4
        class="rs-drawer-title"
      >
        userList.panels.editUser
      </h4>
      <div
        class="rs-drawer-actions"
      >
        <button
          aria-disabled="false"
          class="rs-btn rs-btn-primary"
          data-testid="saveButton"
          type="submit"
        >
          userList.panels.save
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </button>
        <button
          class="rs-btn rs-btn-subtle"
          type="button"
        >
          userList.panels.close
          <span
            class="rs-ripple-pond"
          >
            <span
              class="rs-ripple"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="rs-drawer-body"
    >
      <div
        class="rs-panel"
      >
        <div
          class="rs-panel-body"
          role="region"
        >
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="firstName"
              id="firstName-control-label"
            >
              userList.panels.firstName
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="firstName-help-text"
                aria-labelledby="firstName-control-label"
                class="rs-input"
                id="firstName"
                name="firstName"
                type="text"
                value="Peter"
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="name"
              id="name-control-label"
            >
              userList.panels.name*
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="name-help-text"
                aria-labelledby="name-control-label"
                class="rs-input"
                id="name"
                name="name"
                type="text"
                value="Parker"
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.preferredName
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="preferredName"
                type="text"
                value="Peter Parker"
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
              for="email"
              id="email-control-label"
            >
              userList.panels.email*
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                aria-describedby="email-help-text"
                aria-labelledby="email-control-label"
                class="rs-input"
                id="email"
                name="email"
                type="text"
                value="peter@parker.com"
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.company
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="company"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.streetAddress
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="streetAddress"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.streetAddress2
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="streetAddress2"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.zipCode
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="zipCode"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.city
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="city"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.country
            </label>
            <div
              class="rs-form-control rs-form-control-wrapper"
            >
              <input
                class="rs-input"
                name="country"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.active
            </label>
            <label
              class="rs-toggle"
            >
              <input
                aria-checked="false"
                aria-disabled="false"
                aria-label="Close"
                checked=""
                class="rs-toggle-input"
                role="switch"
                type="checkbox"
              />
              <span
                class="rs-toggle-presentation"
              >
                <span
                  class="rs-toggle-inner"
                />
              </span>
            </label>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <button
              class="rs-btn rs-btn-primary"
              type="button"
            >
              userList.panels.resetPassword
              <span
                class="rs-ripple-pond"
              >
                <span
                  class="rs-ripple"
                />
              </span>
            </button>
            <button
              aria-disabled="true"
              class="rs-btn rs-btn-primary rs-btn-disabled"
              disabled=""
              style="margin-left: 20px;"
              type="button"
            >
              userList.panels.sendWebsiteLogin
              <span
                class="rs-ripple-pond"
              >
                <span
                  class="rs-ripple"
                />
              </span>
            </button>
          </div>
          <div
            class="rs-form-group"
            role="group"
          >
            <label
              class="rs-form-control-label"
            >
              userList.panels.userRoles
            </label>
            <div
              class="rs-picker rs-picker-check rs-picker-default rs-picker-toggle-wrapper rs-picker-placement-auto rs-picker-has-value rs-picker-cleanable rs-picker-block rs-picker-countable"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="rs-picker-toggle rs-btn rs-btn-default"
                role="combobox"
                tabindex="0"
              >
                <input
                  aria-hidden="true"
                  class="rs-picker-toggle-textbox rs-picker-toggle-read-only"
                  readonly=""
                  tabindex="-1"
                  value="roleId1"
                />
                <span
                  class="rs-picker-toggle-value"
                >
                  <span
                    class="rs-picker-value-list"
                    title="Role 1"
                  >
                    <span
                      class="rs-picker-value-item"
                    >
                      Role 1
                    </span>
                  </span>
                  <span
                    class="rs-picker-value-count"
                    title="1"
                  >
                    1
                  </span>
                </span>
                <span
                  aria-label="Clear"
                  class="rs-picker-toggle-clean rs-btn-close"
                  role="button"
                  tabindex="-1"
                  title="Clear"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="close"
                    class="rs-icon"
                    data-category="application"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M2.784 2.089l.069.058 5.146 5.147 5.146-5.147a.5.5 0 01.765.638l-.058.069L8.705 8l5.147 5.146a.5.5 0 01-.638.765l-.069-.058-5.146-5.147-5.146 5.147a.5.5 0 01-.765-.638l.058-.069L7.293 8 2.146 2.854a.5.5 0 01.638-.765z"
                    />
                  </svg>
                </span>
                <svg
                  aria-hidden="true"
                  aria-label="angle down"
                  class="rs-picker-toggle-caret rs-icon"
                  data-category="legacy"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 21 32"
                  width="1em"
                >
                  <path
                    d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
